packages/amplication-server/src/core/resource/resource.service.ts

```diff
@@ -529,266 +546,282 @@ export class ResourceService {
+              const relatedEntityId = (field.properties as unknown as Lookup)
+                .relatedEntityId;
+
+              const fieldProperties = field.properties as unknown as Lookup;
+
+              //If the related entity is moved to the SAME RESOURCE - we should keep the relation
+              if (
+                entities.find((entity) => entity.entityId === relatedEntityId)
+              ) {
+                if (lookupFieldsToSkip.has(field.permanentId)) {
+                  //The other side of this relation field was already moved
+                  continue;
+                }
+
+                const relatedField = await this.entityService.getField({
+                  where: {
+                    permanentId: fieldProperties.relatedFieldId,
+                  },
+                });
+
+                await actionContext.onEmitUserActionLog(
+                  `moving relation field ${field.name} to entity ${currentEntityCreateInput.name} - related field ${relatedField.name} is in the same resource`,
+                  EnumActionLogLevel.Info
+                );
+
+                //@todo: what about the rest of the properties of the related field (like searchable, unique, required, description, customAttributes etc.)
+                createFieldInput.relatedFieldName = relatedField.name;
+                createFieldInput.relatedFieldDisplayName =
+                  relatedField.displayName;
+                createFieldInput.relatedFieldAllowMultipleSelection = (
+                  relatedField.properties as unknown as Lookup
+                ).allowMultipleSelection;
+
+                (
+                  createFieldInput.properties as unknown as Lookup
+                ).relatedFieldId = undefined; //clear the related field id, because it will be set later
+
+                (
+                  createFieldInput.properties as unknown as Lookup
+                ).relatedEntityId = sourceEntityIdToNewEntityMap.get(
+                  fieldProperties.relatedEntityId
+                ).id; //the new entity Id of the related entity
+
+                lookupFieldsToSkip.add(relatedField.permanentId);
+              } else {
+                await actionContext.onEmitUserActionLog(
+                  `moving relation field ${field.name} to resource ${resourceId} - related entity ${relatedEntityId} is in a different resource`,
+                  EnumActionLogLevel.Info
+                );
+
+                //if the related entity is moved to a different resource - we should remove the relation
+                if (!fieldProperties.allowMultipleSelection) {
+                  createFieldInput.name = `${createFieldInput.name}Id`;
+                  createFieldInput.displayName = `${createFieldInput.displayName} ID`;
+                  createFieldInput.dataType = EnumDataType.SingleLineText; //@todo: set the new type based on the ID type of the related entity (WholeNumber or SingleLineText)
```

Comment:
This @todo (setting the ID type) is potentially a blocker for the beta (@PazYanoverr  @yuval-hazaz )

Reply:
Opened https://github.com/amplication/amplication/issues/7934 to track this. Resolving the comment.